.github/workflows/build.yml

```diff
@@ -73,12 +93,14 @@ jobs:
         shell: bash
       - uses: actions/upload-artifact@v2
         with:
-          name: wheels
-          path: target/wheels/*.whl
+          name: dist
+          path: |
+            target/wheels/*.whl
+            target/wheels/*.tar.gz
   publish:
     runs-on: ubuntu-latest
-    needs: "build-rust"
-    if: (github.ref=='refs/heads/main' || github.ref=='refs/heads/dev') && needs.build-rust.result=='success'
```

Comment:
we won't need a successful run because the build will fail if anything in the matrix fails - in theory at least